app/objects/c_agent.py

```diff
@@ -33,11 +33,13 @@ class AgentFieldsSchema(ma.Schema):
     host = ma.fields.String()
     watchdog = ma.fields.Integer()
     contact = ma.fields.String()
+    gui_selected_contact = ma.fields.String()
```

Comment:
let's drop reference to a Gui in a base class object. We want to keep the layers decoupled -- we can think of it just as the 
current, target, and available contacts or similar

Reply:
so the gui_selected_contact field is used to check if the agent has a pending C2 contact switch (I named it that way since the user selects it from the gui). I could rename the field to `pending_contact` - that way it keeps its original purpose, but we don't have gui "references" hanging around in the agent object. Does that sound alright?

Reply:
exactly pending_contact, target_contact, etc. yes same meaning but disconnects from the Gui

Reply:
switched to "pending_contact"